sentinelhub/data_sources.py

```diff
@@ -0,0 +1,522 @@
+"""
+Module defining data sources
+"""
+import re
+import warnings
+from enum import Enum, EnumMeta
+from typing import Tuple
+from dataclasses import dataclass, asdict
+
+from aenum import extend_enum
+
+from .config import SHConfig
+from .constants import ServiceUrl
+from .exceptions import SHDeprecationWarning
+
+
+class _Source:
+    """ Types of satellite sources
+    """
+    SENTINEL2 = 'Sentinel-2'
+    SENTINEL1 = 'Sentinel-1'
+    LANDSAT8 = 'Landsat 8'
+    MODIS = 'MODIS'
+    DEM = 'Mapzen DEM'
+    BYOC = 'BYOC'
+    BATCH = 'BATCH'
+    LANDSAT5 = 'Landsat 5'
+    LANDSAT7 = 'Landsat 7'
+    SENTINEL3 = 'Sentinel-3'
+    SENTINEL5P = 'Sentinel-5P'
+    ENVISAT_MERIS = 'Envisat Meris'
+
+
+class _SensorType:
+    """ Types of satellite sensors
+    """
+    # pylint: disable=invalid-name
+    MSI = 'MSI'
+    IW = 'IW'
+    EW = 'EW'
+    OLCI = 'OLCI'
+    SLSTR = 'SLSTR'
+    TROPOMI = 'TROPOMI'
+
+
+class _ProcessingLevel:
+    """ Types of processing level
+    """
+    # pylint: disable=invalid-name
+    L1B = 'L1B'
+    L1C = 'L1C'
+    L2A = 'L2A'
+    L3B = 'L3B'
+    L2 = 'L2'
+    GRD = 'GRD'
+    MCD43A4 = 'MCD43A4'
+
+
+class _Polarization:
+    """ Types of SAR polarizations
+    """
+    # pylint: disable=invalid-name
+    DV = 'DV'
+    DH = 'DH'
+    SV = 'SV'
+    SH = 'SH'
+
+
+class _Resolution:
+    """ Types of product resolution (specific to Sentinel-1 sources)
+    """
+    MEDIUM = 'MEDIUM'
+    HIGH = 'HIGH'
+
+
+class OrbitDirection:
+    """ Types of orbit directions
+    """
+    ASCENDING = 'ASCENDING'
+    DESCENDING = 'DESCENDING'
+    BOTH = 'BOTH'
+
+
+class _Bands:
+    """ Different collections of bands
```

Comment:
Why are only some bands listed here? E.g. for S2L1C, there are [several other bands available on SH](https://docs.sentinel-hub.com/api/latest/data/sentinel-2-l1c/#available-bands-and-data).

Reply:
I wanted to leave the other "bands" for users to define themselves. They are either metadata bands or products (cloud masks).